src/pages/token/rewards/RewardHistoryPanel.tsx

```diff
@@ -33,17 +33,14 @@ export const RewardHistoryPanel = () => {
     abacusStateManager.getHistoricalTradingRewardPeriod() || HistoricalTradingRewardsPeriod.WEEKLY
   );
 
-  const onSelectPeriod = useCallback(
-    (periodName: string) => {
-      const selectedPeriod =
-        HISTORICAL_TRADING_REWARDS_PERIODS[
-          periodName as keyof typeof HISTORICAL_TRADING_REWARDS_PERIODS
-        ];
-      setSelectedPeriod(selectedPeriod);
-      abacusStateManager.setHistoricalTradingRewardPeriod(selectedPeriod);
-    },
-    [setSelectedPeriod, selectedPeriod]
-  );
+  const onSelectPeriod = useCallback((periodName: string) => {
+    const thisSelectedPeriod =
+      HISTORICAL_TRADING_REWARDS_PERIODS[
+        periodName as keyof typeof HISTORICAL_TRADING_REWARDS_PERIODS
+      ];
+    setSelectedPeriod(thisSelectedPeriod);
+    abacusStateManager.setHistoricalTradingRewardPeriod(thisSelectedPeriod);
+  }, []);
```

Comment:
hmmm looks like `exhaustive-hook-deps` didn't complain about this one?

Reply:
Yep, setState functions are handled correctly.